### PR TITLE
Append Unique Suggestions

### PR DESCRIPTION
**Pull request by jiayinghsu**
### Description
This PR appends labeling suggestions instead of replacing them when the user clicks "Generate Suggestions".  To ensure labeling suggestions are unique, we add a filter at the end of each suggestions method that gets rid of any non-unique suggestions. For the stride and random sampling, we create a unique sample space consisting of the frames not yet in the labeling suggestions. This will inevitably slow down suggestion generation (due to looping through already existing suggestions) - a good idea might be to create a cache for labeling suggestions similar to the `LabelsDataCache`.

### Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
- #838 

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/wiki/Developer-Guide) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [x] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [x] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

File: tests/gui/test_suggestions.py [hunk 1]
```diff
@@ -146,3 +148,197 @@ def test_video_selection(centered_pair_predictions: Labels):
...
+    # only appending unique suggestions
+
+    skeleton = Skeleton()
+    skeleton.add_node("a")
+    skeleton.add_node("b")
+    skeleton.add_node("c")
+
+    track_a = Track(0, "a")
+    # track_b = Track(0, "b")
+    # track_c = Track(0, "c")
```
Comment (roomrys) [inline tests/gui/test_suggestions.py:202]: Remove

**Review by roomrys — CHANGES_REQUESTED** [review]
Some superficial changes requested for the first round

**Review by roomrys — CHANGES_REQUESTED** [review]
Should be using a set difference '-' instead of symmetric difference `^`. Also, we need to account for different videos in `image_feature_based_method`.

**Review by roomrys — APPROVED** [review]
looks good to me

**Review by talmo — APPROVED** [review]
LGTM!

**Pull request merged**